### ISSUE TITLE
Is there any problem to change model float32 to float16

### ISSUE THREAD

**Issue by haonanzc** [idmjky/EvolvePro#9]
By changing the model float32 to float16 like below, I could generate embeddings on my local Mac with 32GB RAM and using esm2_t48_15B_UR50D, but if this change could harm the evolvepro result as the accuracy is decreased?

esm/extract.py
def run(args):
model, alphabet = pretrained.load_model_and_alphabet(args.model_location)
model = model.half() #change float32 to float16, decrease the memory use
model.eval()

**Comment by idmjky** [comment]
Hi, I have not tested the difference between float32 to float16 on its final performance in terms of prediction accuracy. My guess is that this won't hurt too much. You could try comapring this on a few proteinGym datasets to see.
